src/control/tools/StrokeStabilizer.h

```diff
@@ -0,0 +1,666 @@
+/*
+ * Xournal++
+ *
+ * Handles input of strokes
+ *
+ * @author Xournal++ Team
+ * https://github.com/xournalpp/xournalpp
+ *
+ * @license GNU GPLv2 or later
+ */
+#pragma once
+
+#include <array>
+#include <cmath>
+#include <deque>
+#include <functional>
+
+// #include <boost/circular_buffer.hpp>
+
+#include "control/tools/StrokeHandler.h"
+
+#include "StrokeStabilizerEnum.h"
+
+// #define STAB_DEBUG
+
+namespace StrokeStabilizer {
+
+/**
+ * Auxiliary structures
+ */
+
+/**
+ * @brief A (rudimentary) structure for 2D mathematical vectors
+ */
+struct MathVect {
+    double dx{};
+    double dy{};
+    static inline double scalarProduct(MathVect u, MathVect v) { return u.dx * v.dx + u.dy * v.dy; }
+    inline double norm() { return hypot(dx, dy); }
```

Comment:
Is `std::hypot` moved into the `StrokeStabilizer` namespace or the global namespace? Seeing that https://github.com/xournalpp/xournalpp/pull/2468 exists, I'd just like to check that this isn't from a `using namespace std;`. :)

Reply:
It's actually from `#include <cmath>` and not part of the `std` namespace.

Reply:
`std::hypot` is actually in both namespaces. For convenience and since the c++ STL adds some overloads it would be good you'll use the version from the std namespace

Reply:
You're right. I mixed things up: `std::hypot` is from `<cmath>` while the `hypot` I use comes from `mathcalls.h`. I'll change those. Thanks!

Reply:
Done. And rebased onto master